test/user_test.rb

```diff
@@ -0,0 +1,49 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+
+  def setup
+    @sal = User.new("Sal")
+    @ali = User.new("Ali")
+    @joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    @joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+  end
+
+  def test_it_exists
+    assert_instance_of User, @sal
+  end
+
+  def test_user_has_name
+    assert_equal "Sal", @sal.name
+  end
+
+  def test_user_has_jokes
+    assert_equal [], @sal.jokes
+  end
+
+  def test_sal_can_learn_jokes
+    @sal.learn(@joke_1)
+    @sal.learn(@joke_2)
+    assert_equal [@joke_1, @joke_2], @sal.jokes
+  end
+
+  def test_joke_can_be_told
+    @sal.tell(@ali, @joke_1)
+    @sal.tell(@ali, @joke_2)
+    assert_equal [@joke_1, @joke_2], @ali.jokes
+  end
+
+  def test_joke_by_id
+    @sal.tell(@ali, @joke_1)
+    @sal.tell(@ali, @joke_2)
+    assert_equal [@joke_1, @joke_2], @ali.jokes
+    assert_equal [@joke_1], @ali.joke_by_id(1)
+    assert_equal [@joke_2], @ali.joke_by_id(2)
+  end
```

Comment:
this setup works, but would be easier if you only kept line 40 and 43 and changed line 43 to be `@sal` instead of `@ali`